### DISCUSSION TITLE
Loosen requirements

### DISCUSSION THREAD

**Discussion by mahrz24** [reflex-dev/reflex#2770] [Ideas]
I really like the looks of reflex dev so far and would like to use it at my company for internal dev tooling to replace streamlit as it would serve the purpose much better.

However, this is currently made impossible by the quite strict down pinning of a lot of dependencies. As we develop the tools in a larger repository I end up in a lot of conflicts. But even with a standalone tool our security policies frequently require me to update to the latest versions of packages that fix some CVEs or other security issues.

While I can see the case for semantic versioning and down pinning to the major version. In reality this only works in a library if you frequently evaluate if there are newer compatible version and are very quick to lift upper bounds. Sadly with reflex this isn't the case. Maybe a better approach would be to open up most upper boundaries and rely on CI pipelines to ensure compatibility with latest version of libraries.

**Comment by picklelo** [comment]
@mahrz24 Thanks for bringing this up - we definitely want to increase compatibility with other libraries, I created an issue to track this here: #2777

**Reply by mahrz24** [reply]
Thanks for opening up an issue, I wasn't sure if this is something you want to address. Therefore, I opened up a discussion first. 

I created a fork and tried to lift all upper bounds and it looks like this works on os x for unit & integration tests with a small minor change.

Will comment on the issue if I find something out.
